Tasks/Maven/maventask.ts

```diff
@@ -202,9 +202,6 @@ async function execBuild() {
             if (userRunFailed || codeAnalysisFailed) {
                 tl.setResult(tl.TaskResult.Failed, "Build failed."); // Set task failure
             }
-            else {
-                tl.setResult(tl.TaskResult.Succeeded, "Build Succeeded."); // Set task success
-            }
```

Comment:
It seems to me that you're assuming that the task will succeed unless tl.setResult intervenes. Would the code be easier to read / maintain if: 

 when the QG fails we throw an error and have the associated flag set to false ?